internal/lib/lib.go

```diff
@@ -790,28 +791,36 @@ func GetVipNetworkList() []akov1beta1.AviInfraSettingVipNetwork {
 	return VipNetworkList
 }
 
+var vipInfraSyncMap sync.Map
+
 func SetVipInfraNetworkList(infraName string, vipNetworks []akov1beta1.AviInfraSettingVipNetwork) {
-	if VipInfraNetworkList == nil {
-		VipInfraNetworkList = make(map[string][]akov1beta1.AviInfraSettingVipNetwork)
-	}
-	VipInfraNetworkList[infraName] = vipNetworks
+	vipInfraSyncMap.Store(infraName, vipNetworks)
 }
 
 func GetVipInfraNetworkList(infraName string) []akov1beta1.AviInfraSettingVipNetwork {
-	return VipInfraNetworkList[infraName]
+	val, present := vipInfraSyncMap.Load(infraName)
+	if present {
+		return val.([]akov1beta1.AviInfraSettingVipNetwork)
+	}
+	utils.AviLog.Warnf("Key: %s not found in VIP InfraNetwork", infraName)
```

Comment:
In case VIIP Network is not found we are sending an empty slice - can it not cause Avi to pick any network in the IPAM for VIP allocation? should we fallback to the config map's VIP network here?

Reply:
Thats' good point. Will check on that.